spec/expense_spec.rb

```diff
@@ -0,0 +1,70 @@
+require 'rails_helper'
+RSpec.describe 'Expense page', type: :system do
+  before :all do
+    unless User.find_by(email: 'test@gmail.com')
+      new_user = User.new(name: 'test', email: 'test@gmail.com', password: '123456', password_confirmation: '123456')
+      new_user.save!
+    end
+  end
+  describe 'Expense page' do
+    it 'have right content' do
+      unless Category.all.length >= 1
+        new_user = User.find_by(email: 'test@gmail.com')
+        new_category = Category.new(name: 'Food',
+                                    icon: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQmk7E-ji43uI4gdQcYpGO1LLeFFcZ8m9NOTw&usqp=CAU', user: new_user)
+        new_category.save!
+      end
+
+      visit new_user_session_path
+      within('#new_user') do
+        fill_in 'example@gmail.com', with: 'test@gmail.com'
+        fill_in 'Your Password', with: '123456'
+      end
+      click_button 'Log in'
+      click_link('Food')
+      expect(page).to have_content('TRANSACTIONS')
+      expect(page).to have_content('ADD A NEW TRANSACTION')
+      expect(page).to have_content('Total amount')
+      expect(page).to have_content('Most recent')
+    end
+    it 'Display data for concerned transactions' do
+      unless Category.all.length >= 1
+        new_user = User.find_by(email: 'test@gmail.com')
+        new_category = Category.new(name: 'Food',
+                                    icon: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQmk7E-ji43uI4gdQcYpGO1LLeFFcZ8m9NOTw&usqp=CAU', user: new_user)
+        new_category.save!
+      end
+      unless Expense.find_by(name: 'restaurant')
+        new_user = User.find_by(email: 'test@gmail.com')
+        new_transaction = Expense.new(name: 'restaurant', user: new_user, amount: 25)
+        new_transaction.save!
+
+      end
+      unless OperationType.all.length >= 1
+        new_category = Category.find_by(name: 'Food')
+        transaction = Expense.find_by(name: 'restaurant')
+        transaction.create_category_type([new_category.id])
+      end
+      visit new_user_session_path
+      within('#new_user') do
+        fill_in 'example@gmail.com', with: 'test@gmail.com'
+        fill_in 'Your Password', with: '123456'
+      end
+      click_button 'Log in'
+      click_link('Food')
+      expect(page).to have_content('restaurant')
+      expect(page).to have_content('$25.0')
+    end
+    it 'When the user clicks on the "Back" button (<), the user navigates to the home page.' do
+      visit new_user_session_path
+      within('#new_user') do
+        fill_in 'example@gmail.com', with: 'test@gmail.com'
+        fill_in 'Your Password', with: '123456'
+      end
+      click_button 'Log in'
+      click_link('Food')
+      click_link('back')
+      expect(page).to have_current_path(root_path)
+    end
```

Comment:
- [ ] This test fails. Please consider looking into it. See below:
  
  ![image](https://user-images.githubusercontent.com/45002654/159048093-abfc9a2b-96d0-4597-9f73-fb5b1675972c.png)

Reply:
Hi Lameck,
All the test pass successfully in my system . I don't know why it is not working for you.
![Capture d’écran de 2022-03-18 18-05-18](https://user-images.githubusercontent.com/85107514/159049985-820fd758-35bb-485c-b6a3-d7da3d1f7cfc.png)

Reply:
This might be a case of the environment in which the test has been run. I have made this optional.

### TO THE NEXT REVIEWER
This requirement doesn't hold in the next review. The problem has been properly discussed already.